### PR TITLE
switch to 64-bit compilation; add flag for 32-bit

### PR DESCRIPTION
**Pull request by apoelstra**

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
Ugh, can we not do this with cfg(target_pointer_width = "64") and just assume the compiler has __uint128_t?

**Comment by apoelstra** [comment]
Nice, I didn't know about that `cfg` option.

As for `__uint128_t`, I can add a dummy C file that uses it, attempt compilation, and then decide whether it's available based on whether that passes or fails. This is what autotools does.

In fact, maybe that check is sufficient, and I don't even need to determine whether we're on a 64 or 32 bit system?

**Comment by TheBlueMatt** [comment]
The secp autotools stuff still picks the field based on 32 or 64 bit pointer size, independant of __uint128_t, though that's actually probably wrong, we want it based on word size, which target_pointer_width is only a kinda-shitty proxy for. Anyway, I'd say use both?

**Comment by apoelstra** [comment]
Ok, updated to use both, rather than feature flags.

It would be cool if we could trigger this on Travis.

**Comment by TheBlueMatt** [comment]
Yea i was looking into that, apparently the officially recommended way to do that is to spin up a docker container with the desired host. We could also cross-compile to 32 bit and run, I think, as well as cross-compile to Windows and run in wine.

**Comment by apoelstra** [comment]
Tagging this 0.11 because I think we need better test infrastructure to confidently put it in a minor release

File: build.rs [hunk 1]
```diff
@@ -24,6 +24,19 @@
 extern crate cc;
 
 fn main() {
+    // Check whether we can use 64-bit compilation
+    #[cfg(target_pointer_width = "64")]
+    let use_64bit_compilation = {
+        cc::Build::new().file("depend/check_uint128_t.c")
+                        .cargo_metadata(false)
+                        .try_compile("check_uint128_t")
+                        .is_ok()
```
Comment (TheBlueMatt) [inline build.rs]: Do we want to print some kind of warning in case this fails?

**Pull request merged**